lib/hb/gold/track/RandomizedTrack.py

```diff
@@ -1,95 +1,143 @@
-#Untested, to be rewritten
-#import stats
-from gold.util.RandomUtil import random
-import numpy
-from numpy import array
-from gold.track.TrackView import TrackView
+from abc import ABCMeta, abstractmethod
+
+from gold.formatconversion.FormatConverter import TrivialFormatConverter
 from gold.statistic.RawDataStat import RawDataStat
 from gold.track.Track import Track
 from gold.track.TrackFormat import NeutralTrackFormatReq
-from test.gold.track.common.SampleTrackView import SampleTV_Num
-from gold.util.CommonFunctions import isIter
-from gold.util.CustomExceptions import AbstractClassError
-from config.Config import DebugConfig
+from gold.util.CustomExceptions import NotSupportedError
+from quick.util.CommonFunctions import prettyPrintTrackName
+
+
+class TrackRandomizer(object):
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    @abstractmethod
+    def supportsTrackFormat(cls, origTrackFormat):
+        pass
+
+    @classmethod
+    @abstractmethod
+    def supportsOverlapMode(cls, allowOverlaps):
+        pass
+
+    @classmethod
+    def getDescription(cls):
+        return cls.__name__
+
+
+class RandomizedTrack(Track, TrackRandomizer):
+    __metaclass__ = ABCMeta
 
-class RandomizedTrack(Track):
     IS_MEMOIZABLE = False
     WORKS_WITH_MINIMAL = True
 
     def __new__(cls, *args, **kwArgs):
         return object.__new__(cls)
 
     def __init__(self, origTrack, randIndex, **kwArgs):
-        self._origTrack = origTrack
+        self._origTrack = OrigTrackWrapper(origTrack, trackRandomizer=self)
```

Comment:
I agree in general, but very hard to avoid it with this solution. The methods of self that is accessed from OrigTrackWrapper are just simple validation methods which will not use self._origTrack, so should not be a problem. And if one gets infinite recursion due to incorrect code later, it should be easy to spot and fix anyway.